### ISSUE TITLE
Cannot Reply to messages (Exchange account)

### ISSUE THREAD

**Issue by A-Swafford** [thunderbird-conversations/thunderbird-conversations#2132]
* **Thunderbird version:128.1.0esr**
* **Thunderbird Conversation version:4.2.1**
* **Operating System: Latest Mac OS**

**What did you do?**
Tried to reply to an email. No email forwarding, reply, or reply all buttons work. Attempted to disable the quick reply inline. Those two buttons disappeared but the smaller buttons did not work. Right clicking the message in the inbox pane so I could click the normal reply button created a new window with the ability to reply.

**What did you expect to happen?**

I expected to be able to reply to the message

**What actually happened?**
Nothing. It seemed not to treat the buttons as buttons.

**Comment by cs96and** [comment]
I just upgraded to 128.1.0esr on Debian Linux and am having the same issue.  I'm using the Owl extension to access my Exchange account.

**Comment by Standard8** [comment]
Is there anything raised on the developer tools when you click the button?

To get to the console - go to the application menu (three bars) -> Tools -> Developer Tools. Then accept the connection at the prompt, and select the "Console" tab. Also select the "Multiprocess" option at the top.

**Comment by Antidote00** [comment]
I have the same problem under Windows 11. I have to make a right-click on the mail and open in "classic view". This happens ONLY on exchange accounts using the "Eule für Exchange" Plugin. Normal IMAP or POP work just fine. 

Here is the console log from clicking a button:

`Uncaught (in promise) TypeError: defaultAccount is null
    getBestIdentityForReply chrome://conversations/content/164.bundle.js:4809
    expand chrome://conversations/content/164.bundle.js:5137
    middleware chrome://conversations/content/251.bundle.js:1934
    expand chrome://conversations/content/164.bundle.js:3885
    Nb chrome://conversations/content/537.bundle.js:172
    Tb chrome://conversations/content/537.bundle.js:172
    Ub chrome://conversations/content/537.bundle.js:173
    nf chrome://conversations/content/537.bundle.js:223
    se chrome://conversations/content/537.bundle.js:224
    hd chrome://conversations/content/537.bundle.js:235
    Qk chrome://conversations/content/537.bundle.js:391
    Jb chrome://conversations/content/537.bundle.js:170
    hd chrome://conversations/content/537.bundle.js:227
    fd chrome://conversations/content/537.bundle.js:192
    ed chrome://conversations/content/537.bundle.js:191
    pf chrome://conversations/content/537.bundle.js:226
    qf chrome://conversations/content/537.bundle.js:225
    sf chrome://conversations/content/537.bundle.js:225
    sf chrome://conversations/content/537.bundle.js:225
    createRoot chrome://conversations/content/537.bundle.js:436
    591 chrome://conversations/content/stub.bundle.js:2827
    591 chrome://conversations/content/stub.bundle.js:2811
    __webpack_require__ chrome://conversations/content/stub.bundle.js:2865
    __webpack_exports__ chrome://conversations/content/stub.bundle.js:3023
    O chrome://conversations/content/stub.bundle.js:2899
    <anonymous> chrome://conversations/content/stub.bundle.js:3024
    <anonymous> chrome://conversations/content/stub.bundle.js:3026
2 164.bundle.js:4809:11
Uncaught (in promise) TypeError: defaultAccount is null
    getBestIdentityForReply chrome://conversations/content/164.bundle.js:4809
    expand chrome://conversations/content/164.bundle.js:5137
    middleware chrome://conversations/content/251.bundle.js:1934
    expand chrome://conversations/content/164.bundle.js:3883
    Nb chrome://conversations/content/537.bundle.js:172
    Tb chrome://conversations/content/537.bundle.js:172
    Ub chrome://conversations/content/537.bundle.js:173
    nf chrome://conversations/content/537.bundle.js:223
    se chrome://conversations/content/537.bundle.js:224
    hd chrome://conversations/content/537.bundle.js:235
    Qk chrome://conversations/content/537.bundle.js:391
    Jb chrome://conversations/content/537.bundle.js:170
    hd chrome://conversations/content/537.bundle.js:227
    fd chrome://conversations/content/537.bundle.js:192
    ed chrome://conversations/content/537.bundle.js:191
    pf chrome://conversations/content/537.bundle.js:226
    qf chrome://conversations/content/537.bundle.js:225
    sf chrome://conversations/content/537.bundle.js:225
    sf chrome://conversations/content/537.bundle.js:225
    createRoot chrome://conversations/content/537.bundle.js:436
    591 chrome://conversations/content/stub.bundle.js:2827
    591 chrome://conversations/content/stub.bundle.js:2811
    __webpack_require__ chrome://conversations/content/stub.bundle.js:2865
    __webpack_exports__ chrome://conversations/content/stub.bundle.js:3023
    O chrome://conversations/content/stub.bundle.js:2899
    <anonymous> chrome://conversations/content/stub.bundle.js:3024
    <anonymous> chrome://conversations/content/stub.bundle.js:3026
2 164.bundle.js:4809:11
Uncaught (in promise) TypeError: defaultAccount is null
    getBestIdentityForReply chrome://conversations/content/164.bundle.js:4809
    getParamsForCompose chrome://conversations/content/stub.bundle.js:36
    reply chrome://conversations/content/stub.bundle.js:102
    middleware chrome://conversations/content/251.bundle.js:1934
    onActionButtonClick chrome://conversations/content/164.bundle.js:2394
    action chrome://conversations/content/164.bundle.js:472
    Nb chrome://conversations/content/537.bundle.js:172
    Tb chrome://conversations/content/537.bundle.js:172
    Ub chrome://conversations/content/537.bundle.js:173
    nf chrome://conversations/content/537.bundle.js:223
    se chrome://conversations/content/537.bundle.js:224
    hd chrome://conversations/content/537.bundle.js:235
    Qk chrome://conversations/content/537.bundle.js:391
    Jb chrome://conversations/content/537.bundle.js:170
    hd chrome://conversations/content/537.bundle.js:227
    fd chrome://conversations/content/537.bundle.js:192
    ed chrome://conversations/content/537.bundle.js:191
    pf chrome://conversations/content/537.bundle.js:226
    qf chrome://conversations/content/537.bundle.js:225
    sf chrome://conversations/content/537.bundle.js:225
    sf chrome://conversations/content/537.bundle.js:225
    createRoot chrome://conversations/content/537.bundle.js:436
    591 chrome://conversations/content/stub.bundle.js:2827
    591 chrome://conversations/content/stub.bundle.js:2811
    __webpack_require__ chrome://conversations/content/stub.bundle.js:2865
    __webpack_exports__ chrome://conversations/content/stub.bundle.js:3023
    O chrome://conversations/content/stub.bundle.js:2899
    <anonymous> chrome://conversations/content/stub.bundle.js:3024
    <anonymous> chrome://conversations/content/stub.bundle.js:3026
2 164.bundle.js:4809:11
`

**Comment by Standard8** [comment]
I'm pretty sure that this is an issue with either Thunderbird or the Owl add-on, so I would suggest reporting it to the Owl authors in the first instance. There's a support email link on their [add-on page](https://addons.thunderbird.net/thunderbird/addon/owl-for-exchange/).

What appears to be happening is that the one of the two has potentially broken the accounts and identity system within Thunderbird, so that there is now no default account. When pressing the reply button, Conversations is trying to identify the email address to respond from (i.e. one of your email addresses), and hasn't been able to do that. Worst case, it [falls back to the "default" account](https://github.com/thunderbird-conversations/thunderbird-conversations/blob/c504fa62cc3b9626b8f623ad659fef1a4e1e0709/addon/content/reducer/messageUtils.mjs#L36-L47), but that is failing according to those logs.

As Conversations' is using the standard Thunderbird add-on interfaces (APIs) for these actions, my suspicion is that it is either the Owl add-on or something it is doing within Thunderbird that is causing the Thunderbird APIs to break.

In summary, unfortunately there is nothing I can do to fix this in Conversations.

**Comment by cs96and** [comment]
Unfortunately I cannot get the developer tools to appear (selecting the menu option, or using the keyboard shortcut does nothing).

Owl support cannot reproduce the bug.

**Comment by Standard8** [comment]
> Unfortunately I cannot get the developer tools to appear (selecting the menu option, or using the keyboard shortcut does nothing).

That's strange, I'm not sure why that would be.

> Owl support cannot reproduce the bug.

Given the log we have received, I still think it is the Owl add-on or Thunderbird that are causing this. As I mentioned, Conversations is using the standard code, and doesn't do anything special.

The only other thing I can think of to try and get you working is to go into the Account Settings, select one of your accounts on the left, and then from the "Account Actions" button, choose "Set as Default".

You should be able to set it back again to a different account if you want to afterwards, but it might just be enough to get the settings right in Thunderbird.

**Comment by cs96and** [comment]
I only have one account, so the Set As Default action is greyed out.  I tried adding a dummy account, setting that to default then back to the Owl one, but the problem persists.

However, if I set the dummy account to the default, then the button starts working again.

**Comment by Standard8** [comment]
Okay, I really don't know what that is. I would suggest following up with Owl support, it definitely feels like something caused by their add-on, even if they can't reproduce it. Worst case, try filing a bug with Thunderbird.

**Comment by cs96and** [comment]
I managed to get the console open by starting thunderbird with the `--jsconsole` parameter.  Hitting reply gives the following error...

```
Uncaught (in promise) TypeError: defaultAccount is null
    getBestIdentityForReply chrome://conversations/content/164.bundle.js:4809
```

I looked at the code around that area, I see it is calling `browser.accounts.get` and `browser.accounts.getDefault`.  I tried calling these from the console but I get `Uncaught ReferenceError: browser is not defined`.  Do you know how if it is possible to get access to the `browser` object from the console.

Also, trying to start the Dev Tools when the console is open says `Could not start Browser Toolbox, you need to enable it.`  Do you know how to enable it?

Thanks

**Comment by cs96and** [comment]
Just found this bug report in Thunderbird
https://bugzilla.mozilla.org/show_bug.cgi?id=1909005

**Comment by Standard8** [comment]
Thunderbird 128.2.0 has now been released. I'm told this should fix the issues with Owl.

**Comment by cs96and** [comment]
Confirmed that I can now reply to messages with 128.2.0esr
